app/src/main/java/com/pvp/app/common/Utils.kt

```diff
@@ -44,4 +46,18 @@ fun Picture.toImageBitmap(): ImageBitmap {
 
             return bitmap.asImageBitmap()
         }
+}
+
+fun List<ExerciseSessionRecord>.toSportActivities(): List<SportActivity> {
+    return this.mapNotNull { exercise ->
+        SportActivity.fromId(exercise.exerciseType)
+    }
+}
+
+fun List<SportActivity>.getOccurences(): List<Pair<SportActivity, Int>> {
+    return this
```

Comment:
Also note that you are in `List<SportActivity>` instance context, so you can call all methods of the list without specifying `this`. If you want to use it, let it be. If you don't want to, then it is possible, just call it as so:
```kotlin
    return groupingBy { it }
        .eachCount()
        .toList()
        .sortedByDescending { it.second }
```